pkg/database/database.go

```diff
@@ -48,12 +48,29 @@ func (db *DbInstance) DSN() string {
 	return db.dsn
 }
 
+// DB is a global db instance.
 var DB *DbInstance = &DbInstance{}
 
-func ConnectDB(
+// ConnectDB will establish and return a DbInstance while also caching it in the global
+// var database.DB.
+func ConnectDB(dsn string, slowThreshold time.Duration, poolMax int, reset bool, migrate bool, artifactRoot string,
+) (*DbInstance, error) {
+	db, err := MakeDBInstance(dsn, slowThreshold, poolMax, reset, migrate, artifactRoot)
+	if err != nil {
+		return nil, err
+	}
+	// set the global DB
+	DB = db
+	return DB, nil
+}
+
+// MakeDbInstance will create a DbInstance from the parameters.
+func MakeDBInstance(
```

Comment:
Let's tackle this when we add a third adapter (DuckDB)? Seems out of scope for this story.

Reply:
but I think right now is a really good moment, because we have no implementation at all, so it will be nice to create a good/extendable base for next integrations?

Reply:
this story has been stuck for a while and Dave is on my case, so i don't want to add out-of-scope requirement to refactor the database registration :-) -- and the long-lived branch is harder to maintain.

Reply:
created new ticket for refactoring this https://github.com/G-Research/fasttrackml/issues/244